Python/_warnings.c

```diff
@@ -1298,12 +1298,16 @@ PyErr_WarnExplicit(PyObject *category, const char *text,
                    const char *module_str, PyObject *registry)
 {
     PyObject *message = PyUnicode_FromString(text);
+    if (message == NULL) {
+        goto exit;
+    }
     PyObject *filename = PyUnicode_DecodeFSDefault(filename_str);
+    if (filename == NULL) {
+        goto exit;
+    }
     PyObject *module = NULL;
     int ret = -1;
```

Comment:
You can move this line to 1300 to fix the warning :)

Reply:
Oh, thanks! :) I just removed the variable; IMO that makes the code easier to read.

Reply:
I ended up rewriting the whole PR.